File: src/main/java/com/jacuum/web/SessionApi.java
package com.jacuum.web;

import com.jacuum.engine.SessionView;
import com.jacuum.web.dto.CreateSessionRequest;
import com.jacuum.web.dto.SessionResponse;

interface SessionApi {
    SessionResponse create(CreateSessionRequest req) throws Exception;
    SessionView start(String id) throws Exception;
    SessionView pause(String id) throws Exception;
    SessionView resume(String id) throws Exception;
    SessionView stop(String id) throws Exception;
    SessionView view(String id) throws Exception;
}

File: src/main/kotlin/com/jacuum/web/SessionApi.kt
package com.jacuum.web

import com.jacuum.engine.SessionView
import com.jacuum.web.dto.CreateSessionRequest
import com.jacuum.web.dto.SessionResponse

internal interface SessionApi {
    @Throws(Exception::class) fun create(req: CreateSessionRequest): SessionResponse
    @Throws(Exception::class) fun start(id: String): SessionView
    @Throws(Exception::class) fun pause(id: String): SessionView
    @Throws(Exception::class) fun resume(id: String): SessionView
    @Throws(Exception::class) fun stop(id: String): SessionView
    @Throws(Exception::class) fun view(id: String): SessionView
}
